### PR TITLE
MDS-3846 - separate docman from build.yaml into separate github action

### PR DESCRIPTION
**Pull request by wilson496**
# Main

- What did you focus on in this PR?

Creating a separate GitHub action for document-manager.

# Other

- What other side fixes, tweaks did you add on the side?

document-manager has been removed from the monolithic build.yaml file and is a separate job that can be viewed as an individual GitHub action. 

# How to test

Trigger GitHub action "Build docman" and ensure it succeeds.

NOTE: As of the time of this PR, lint manifests may fail even if the build is successful. Linting for these builds are a W.I.P and do not work correctly. Thus, these can be ignored for the time being according to @jon-funk .  

# Notes

- Any notes/screenshots for this PR?
- N/A

### CODE REVIEW

File: .github/workflows/build-docman.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Build docman
+
+on:
+  pull_request:
+  push:
+    branches:
+      - develop
+    path:
+      - services/document-manager/backend/**
+      - .github/workflows/build-tusd.yaml
```
Comment (jon-funk) [inline .github/workflows/build-docman.yaml]: Should be `build-docman.yaml`

File: .github/workflows/build-docman.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+env:
+  INITIAL_TAG: latest
+  TAG: dev
+  NAME: docman
+  CONTEXT: services/document-manager/backend/
+
+jobs:
+  build-tusd:
```
Comment (jon-funk) [inline .github/workflows/build-docman.yaml]: Rename job to `build-docman`

**Comment by jon-funk** [comment]
We'll want to update https://github.com/bcgov/mds/blob/develop/openshift4/templates/docman.dc.yaml#L344 as well for the worker definition to match the new script location in the image. (the current path is for the redhat image)

**Comment by jon-funk** [comment]
Confirmed that NoW doc uploads are functioning with the new build in test.

File: .github/workflows/build-docman.yaml (hunk 1 above)
Comment (wilson496) [reply .github/workflows/build-docman.yaml]: Oops! I'll get on that.

**Review by jon-funk — APPROVED** [review]
Nice work! 🎉

**Pull request merged**